sympy/sets/sets.py

```diff
@@ -534,6 +535,39 @@ def measure(self):
         """
         return self._measure
 
+    @property
+    def kind(self):
+        """
+        This method gives kind of set and its subclasses,\n
+        Class of kind is ``SetKind``,\n
+        Parenthesis of ``SetKind`` contains kind of element of the set such as ``NumberKind``,\n
+        Example\n
+        kind of Interval(1,2) will ``SetKind(NumberKind)``
+
+        Examples
+        ========
+
+        >>> from sympy import Interval, Matrix, FiniteSet, EmptySet, ProductSet, PowerSet
+        >>> FiniteSet(Matrix([1, 2])).kind
+        SetKind(MatrixKind(NumberKind))
+        >>> FiniteSet(0, Matrix([1, 2])).kind
+        SetKind(UndefinedKind)
```

Comment:
These examples would work better as documentation if there was some explanation in text as well.